wire-ios/Wire-iOS/Sources/Developer/DebugAlert.swift

```diff
@@ -89,19 +89,30 @@ final class DebugAlert {
         controller.present(alert, animated: true, completion: nil)
     }
 
-    static func displayFallbackActivityController(logPaths: [URL],
-                                                  email: String,
-                                                  from controller: UIViewController,
-                                                  sourceView: UIView? = nil) {
+    static func displayFallbackActivityController(
+        email: String,
+        from controller: UIViewController,
+        sourceView: UIView? = nil
+    ) {
         let alert = UIAlertController(
             title: L10n.Localizable.Self.Settings.TechnicalReportSection.title,
             message: L10n.Localizable.Self.Settings.TechnicalReport.noMailAlert + email,
             preferredStyle: .alert
         )
         alert.addAction(.cancel())
         alert.addAction(UIAlertAction(title: L10n.Localizable.General.ok, style: .default, handler: { _ in
-            let activity = UIActivityViewController(activityItems: logPaths, applicationActivities: nil)
+            let logFilesProvider = LogFilesProvider()
+            let logsFileURL = logFilesProvider.generateLogFilesZip()
+
+            let activity = UIActivityViewController(activityItems: [logsFileURL], applicationActivities: nil)
             activity.configPopover(pointToView: sourceView ?? controller.view)
+            activity.completionWithItemsHandler = { _, _, _, _ in
+                do {
+                    try logFilesProvider.clearTemporaryDirectory()
```

Comment:
It is part of the [documentation](https://github.com/wireapp/wire-ios/blob/08680fcc8ecc169798ba0a1c93a1c380c51c3bf8/wire-ios/Wire-iOS/Sources/UserInterface/Settings/Debug%20Report/LogFilesProvider.swift#L42C5-L42C42) of `LogFilesProvider` to clean up:

```swift
/// Clears the temporary directory. 
/// Call once you are done using the URL returned by `generateLogFilesZip` to clean up.
func clearTemporaryDirectory() throws
```

Basically `LogFilesProvider` creates a temporary directory for saving a zip file too. Once we are done with the zip `clearTemporaryDirectory()` removes the temporary directory.

However now you mention it perhaps the behavior of `LogFilesProvider` in it's current implementation may be buggy. 

I _expect_ the following might fail and it may be worth looking into it @David-Henner.

```
let provider = LogFilesProvider() // Implicitly creates temp directory
let a = generateLogFilesZip()
try provider. clearTemporaryDirectory() // Deletes temp directory
let b = generateLogFilesZip() // <- This may now fail without error because the temporary directory no longer exists.
```

Reply:
Also make sure temporary directory is unique as NSTemporaryDirectory is used elsewhere

Reply:
Good point @netbe . No it is not unique! I just tried it. Will mention this in https://github.com/wireapp/wire-ios/pull/1658

Reply:
Please could you re-review @netbe

Reply:
I'm resolving this conversation as I think the points are addressed. Feel free to re-open.